[Roald Ong] iP

SynapseProgramming opened the pull request:
# I present the new chatbot (usagi) うさぎ
> it is simply a rabbit

- it is a bunny
- it likes to hop
1. i like to hop
2. I like to swim
3. I like to fly

```
System.out.println("hop!");
```
``` java
System.out.println("test");
String wow = "wow";
```
this is a link to [google](https://www.google.com/)
- [x] basic version
- [x] gui
 :+1: theres so much code to write :) :-1:

this is   **nice**

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

SynapseProgramming committed:
Added in initial greetings

SynapseProgramming committed:
Added in echo function

SynapseProgramming committed:
Completed level-2 feature

SynapseProgramming committed:
Added in mark command

SynapseProgramming committed:
Modified input string parsing

SynapseProgramming committed:
Added in Task class

SynapseProgramming committed:
Renamed userInputs array as taskList. Added in task marking

SynapseProgramming committed:
Added in unmark function

SynapseProgramming committed:
Converted if-else chain to switch statement

SynapseProgramming committed:
Added in todo command

SynapseProgramming committed:
Added in Todo class to represent Todo event

SynapseProgramming committed:
Added in UiFormatter class for formatting objects of type Task

SynapseProgramming committed:
completed "todo" command

SynapseProgramming committed:
removed default add task, and changed print statements to use UiFormatter

SynapseProgramming committed:
Added in Deadline class

SynapseProgramming committed:
Added in formatting for Deadline objects

SynapseProgramming committed:
Added in deadline command

SynapseProgramming committed:
Added in Event class

SynapseProgramming committed:
Added in Event class to displayTask function

SynapseProgramming committed:
Added in Event command

SynapseProgramming committed:
Added in input testing

SynapseProgramming committed:
Added in exception handling for mark and unmark commands

SynapseProgramming committed:
Added in exception handling for end of line

SynapseProgramming committed:
Added in empty list check for list command

SynapseProgramming committed:
Added in exception handling for todo command

SynapseProgramming committed:
Added in exception handling for deadline command

SynapseProgramming committed:
Added in exception handling for event command

SynapseProgramming committed:
Added in case for deleting entries from the list

SynapseProgramming committed:
Merge pull request #1 from SynapseProgramming/devel

Completed week 2 tasks

SynapseProgramming committed:
Merge branch 'add-gradle-support' into branch-Level-7

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support

SynapseProgramming committed:
Add json library as a dependency

SynapseProgramming committed:
Add in Tasklistloader class

SynapseProgramming committed:
Add in test json file

SynapseProgramming committed:
Enable polymorphism for json Task parser

SynapseProgramming committed:
remove test json file

SynapseProgramming committed:
Modify description field of Task class to private

SynapseProgramming committed:
Fix missing fields in json file

SynapseProgramming committed:
Modify access of description class

SynapseProgramming committed:
Add in task.json test file

SynapseProgramming committed:
Add in first working json serializing/ deserializing attempt

SynapseProgramming committed:
Remove commented code

SynapseProgramming committed:
Code linting

SynapseProgramming committed:
Add in empty constructor required by Jackson library

SynapseProgramming committed:
Remove unused headers and comments

SynapseProgramming committed:
Fix class field visibility for Jackon. & Save previous data to arraylist

SynapseProgramming committed:
change done to boolean object type

SynapseProgramming committed:
Remove print test statements and commented code

SynapseProgramming committed:
Place save tasks function to end of main function

SynapseProgramming committed:
Add in more verbose warning for ( tasks.json not found)

SynapseProgramming committed:
Remove test tasks.json file

SynapseProgramming committed:
Merge pull request #2 from SynapseProgramming/branch-Level-7

Add task save to json file

SynapseProgramming committed:
Add DeadTime support for parser

SynapseProgramming committed:
Change date field to LocalDate type for Deadline class

SynapseProgramming committed:
Add in string formatting to Localdate type for deadline command

SynapseProgramming committed:
Remove toto and test print statements

SynapseProgramming committed:
Merge pull request #3 from SynapseProgramming/branch-Level-8

Change deaddate field from string type to Date type

SynapseProgramming committed:
Rename ReceivedTasks to TaskList

SynapseProgramming committed:
Rename Tasklistloader to Storage

SynapseProgramming committed:
Add in initial parser class

SynapseProgramming committed:
Remove commented code

SynapseProgramming committed:
Add in getIndex function into parser

SynapseProgramming committed:
Add in getIndex function call for unmark command

SynapseProgramming committed:
Add in getCommandString function

SynapseProgramming committed:
Added in getTaskName function

SynapseProgramming committed:
Add getTask name for deadline command

SynapseProgramming committed:
Add in getIndex for remove command

SynapseProgramming committed:
Rename class UiFormatter to Ui

SynapseProgramming committed:
Remove unused variables

SynapseProgramming committed:
Remove unused variables and added in isInputThere function

SynapseProgramming committed:
Change TaskList to be the main array for storing tasks

SynapseProgramming committed:
Change Storage class to handle tasklist

SynapseProgramming committed:
Change taskList to be of type TaskList

SynapseProgramming committed:
Code refactor for main function

SynapseProgramming committed:
Merge pull request #4 from SynapseProgramming/devel

Completed more OOP part.

SynapseProgramming committed:
Move all task related classes to tasks package

SynapseProgramming committed:
change import to get task related stuff from task package

SynapseProgramming committed:
Move storage class to storage package

SynapseProgramming committed:
Move parser and ui class to io folder

SynapseProgramming committed:
update main duke class to import classes from the new packages

SynapseProgramming committed:
Merge pull request #5 from SynapseProgramming/devel

Organised code into packages

SynapseProgramming committed:
add in parseDeadline function, which handles parsing for deadline command.

SynapseProgramming committed:
Add in custom exception for Parser

SynapseProgramming committed:
Integrate parser exception for parseDeadline function

SynapseProgramming committed:
Add in parseEvent function and modified duke run function

SynapseProgramming committed:
Remove unused headers

SynapseProgramming committed:
Add in display greetings function

SynapseProgramming committed:
Add in display action function to ui class

SynapseProgramming committed:
Change print + action statements to use displayAction function.

SynapseProgramming committed:
Add in displayGoodbye function

SynapseProgramming committed:
Add listTasks function

SynapseProgramming committed:
Add markTaskAsDone function

SynapseProgramming committed:
Add unmarkTask function

SynapseProgramming committed:
Add in addTodo function

SynapseProgramming committed:
Add in addDeadline function

SynapseProgramming committed:
Add in addEvent function

SynapseProgramming committed:
Add in deleteTask function

SynapseProgramming committed:
Merge pull request #6 from SynapseProgramming/devel

Code Refactoring and custom  exceptions

SynapseProgramming committed:
Add in J-unit test case for parseEvent function

SynapseProgramming committed:
Add in additional tests for parseDeadline function

SynapseProgramming committed:
Add in UiTest test file

SynapseProgramming committed:
Add in additional tests for displayTask function

SynapseProgramming committed:
Merge pull request #7 from SynapseProgramming/devel

Add in J-unit tests

SynapseProgramming committed:
Move Duke class to another packaged called duke

SynapseProgramming committed:
Add in jar file creation

SynapseProgramming committed:
Merge pull request #8 from SynapseProgramming/devel

Add in .jar file compilation settings, and moved duke class to its own package

SynapseProgramming committed:
Add in javadoc comments for  all parser functions

SynapseProgramming committed:
Add in javadoc comments for  Storage functions

SynapseProgramming committed:
Add in javadoc comments for  Ui class

SynapseProgramming committed:
Add in javadoc comments for all classes

SynapseProgramming committed:
Apply updated code linting

SynapseProgramming committed:
Add in find function

SynapseProgramming committed:
Merge level-9 branch with main

SynapseProgramming committed:
Merge javadoc branch with main

SynapseProgramming commented:
# I present the new chatbot (usagi) うさぎ
> it is simply a rabbit

- it is a bunny
- it likes to hop
1. i like to hop
2. I like to swim
3. I like to fly

```
System.out.println("hop!");
```
``` java
System.out.println("test");
String wow = "wow";
```
this is a link to [google](https://www.google.com/)
- [x] basic version
- [x] gui
 :+1: theres so much code to write :) :-1:

this is   **nice**

SynapseProgramming committed:
stashing here for now

SynapseProgramming committed:
Add in printarray and related functions

SynapseProgramming committed:
Replace print statements with add to string array list

SynapseProgramming committed:
Migrate print statements through UI class

SynapseProgramming committed:
Add in java fx as dependencies

SynapseProgramming committed:
Add in dialog box and main window fxml files

SynapseProgramming committed:
Add in images

SynapseProgramming committed:
Add in fxml related files

SynapseProgramming committed:
Remove main function and fix list function

SynapseProgramming committed:
Remove unused headers

SynapseProgramming committed:
Add in start message

SynapseProgramming committed:
Code linting for storage class

SynapseProgramming committed:
Add in exit functions to bye command

SynapseProgramming committed:
Merge pull request #9 from SynapseProgramming/fxml

Add in fxml code

SynapseProgramming committed:
Code linting

SynapseProgramming committed:
Add in javadoc for print statement.

SynapseProgramming committed:
Merge pull request #10 from SynapseProgramming/Level-10

Replace text based gui with java fx gui

samuelmui8 commented on src/main/java/io/Parser.java:
```diff
@@ -0,0 +1,175 @@
+     *
+     * @throws NoSuchElementException If no further input is available.
+     */
+    public void update() throws NoSuchElementException {
```
can consider a more descriptive name (update what?)

samuelmui8 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+
+    }
+
+    public String getResponse(String input) {
```
can consider adding javadocs / comments to describe what this method does

yytan25 replied:
Seconded. I like how you used encapsulation to pass the functionality to the `Ui` class, but it is not immediately obvious what this method does.

samuelmui8 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    public static final String taskType = "D";
+
+    private LocalDate deadDate;
```
can consider renaming this field (eg endDate)

samuelmui8 commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+
+    private LocalDate deadDate;
+
+    public Deadline(String desc, LocalDate deaddate) {
```
camelCase not followed here

samuelmui8 reviewed (commented):
Good job as coding standard is pretty solid other than some minor naming issues. Can consider adding javadocs to all your public methods so that it is clear what each of your methods are doing. LGTM! 😸

yytan25 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,190 @@
+
+        parser.update(input);
+        // there is no input
+        switch (parser.getCommandString()) {
+            case "bye":
```
Perhaps you should not indent the `case` clauses?

yytan25 commented on src/main/java/io/Parser.java:
```diff
@@ -0,0 +1,175 @@
+    }
+
+    /**
+     * updates the internal input string function
```
Just a nit here, but perhaps you could capitalize the first word of your Javadoc comment and end the comment with a full stop?

yytan25 commented on src/main/java/io/Ui.java:
```diff
@@ -0,0 +1,110 @@
+public class Ui {
+
+
+    private ArrayList<String> printArray;
```
The name `printArray` would be more applicable for a method that prints an output array. Perhaps a noun, such as `arrayToBePrinted`, would be a more suitable name for your `ArrayList`?

yytan25 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,59 @@
+    @JsonIgnore
+    public List<Task> findTasks(String description) {
+        List<Task> output;
+        output = tasks.stream().filter(task -> task.getDescription().contains(description)).collect(
+            Collectors.toList());
```
Great job for using the Streams function!

yytan25 commented on src/main/java/tasks/Todo.java:
```diff
@@ -0,0 +1,24 @@
+public class Todo extends Task {
+
+
+    public static final String taskType = "T";
```
Perhaps you can use the `SCREAMING_SNAKE_CASE` for your constants (also applies all your related `Task` subclasses), as in `TASK_TYPE`? I understand that Java's CheckStyle might think otherwise though...

yytan25 reviewed (commented):
LGTM apart from a few nits related to coding style. Your product is overall very robust and I like its personality. Well done!

SynapseProgramming committed:
Add in checkstyle tests

SynapseProgramming committed:
Apply correct indentation rules for test code

SynapseProgramming committed:
Merge pull request #11 from SynapseProgramming/checkstyle

Add in checkstyle checks

SynapseProgramming committed:
Set fxgui.Launcher as the main entry class

SynapseProgramming committed:
Separate import groups

SynapseProgramming committed:
Add in assertions to getIndex function

SynapseProgramming committed:
Merge pull request #12 from SynapseProgramming/A-Assertions

Add in assertions check for getIndex function

SynapseProgramming committed:
Update import ordering for storage class

SynapseProgramming committed:
Update import ordering for parser class

SynapseProgramming committed:
Resolve checkstyle issues for Duke class

SynapseProgramming committed:
Add newline for Launcher class

SynapseProgramming committed:
Resolve checkstyle issues with Main class

SynapseProgramming committed:
Resolve checkstyle issues with MainWindow class

SynapseProgramming committed:
Resolve checkstyle issues with Parser class

SynapseProgramming committed:
Resolve checkstyle issues with DeadlineClass

SynapseProgramming committed:
Resolve checkstyle issues with Event Class

SynapseProgramming committed:
Resolve checkstyle issues with Task Class

SynapseProgramming committed:
Resolve checkstyle issues with TaskList class

SynapseProgramming committed:
Resolved checkstyle issues with the Todo class

SynapseProgramming committed:
Merge pull request #13 from SynapseProgramming/A-CodeQuality

Fix most code quality issues

SynapseProgramming committed:
Resolve checkstyle issues with ParserTest

SynapseProgramming committed:
Resolve checkstyle issues with UiTest.java

SynapseProgramming committed:
Add in CI workflow file

SynapseProgramming committed:
Add in gradle test statement into file

SynapseProgramming committed:
Fix filepath and naming issues

SynapseProgramming committed:
Remove ubuntu from platforms

SynapseProgramming committed:
Merge pull request #14 from SynapseProgramming/A-CI

Add in CI github workflow actions file

SynapseProgramming committed:
Change event to use LocalDateTime to represent start and end times

SynapseProgramming committed:
Update unit tests for display task event

SynapseProgramming committed:
Fix checkstyle warnings

SynapseProgramming committed:
Add in checks to ensure that the end date time is later than the start date time

SynapseProgramming committed:
Add in getEvents function to get all events from list

SynapseProgramming committed:
Add in isConflict function, which checks if there is a conflict between two events

SynapseProgramming committed:
Add in conflict checks for adding tasks

SynapseProgramming committed:
Add in start and end date getter functions

SynapseProgramming committed:
Add in freeTime function, which returns the users next available datetime

SynapseProgramming committed:
Merge pull request #15 from SynapseProgramming/B-FindFreeTimes

Completed find free Times functionality

SynapseProgramming committed:
Add in Ui.png photo

SynapseProgramming committed:
Add in delay to bye command

The kill commands are now spun up
in a separate thread

After a 2-second delay, the kill
commands would be called.

SynapseProgramming committed:
Rename done to isDone

SynapseProgramming committed:
Merge pull request #17 from SynapseProgramming/final-edits

Add sleep timer and edited boolean variable name

SynapseProgramming committed:
Update README.md

- first version of README

SynapseProgramming committed:
Update README.md

- Add in tasks description

SynapseProgramming committed:
initial version of user guide

SynapseProgramming committed:
Formatting and fix errata for the user guide.

LuahJunYang commented on src/main/java/tasks/Event.java:
```diff
@@ -0,0 +1,75 @@
+     */
+    @JsonIgnore
+    public boolean isConflict(Event e) {
+        if (startDate.isAfter(e.startDate) && startDate.isBefore(e.endDate) && endDate.isAfter(
+            e.endDate)) {
+            return true;
+        } else if (startDate.isBefore(e.startDate) && endDate.isAfter(e.endDate)) {
+            return true;
+        } else if (startDate.isBefore(e.startDate) && endDate.isAfter(e.startDate)) {
+            return true;
+        } else if (startDate.isAfter(e.startDate) && endDate.isBefore(e.endDate)) {
+            return true;
+        }
```
Your overall code is very readable and nicely formatted, perhaps you could consider using the switch statement here instead of the multiple if-else statements.

LuahJunYang reviewed (commented):
Amazing job done for the IP.
Just one minor suggestion as commented.
